src/DynamoCoreWpf/Extensions/ViewLoadedParams.cs

```diff
@@ -69,14 +69,24 @@ public Window DynamoWindow
             dynamoMenu = dynamoView.titleBar.ChildOfType<Menu>();
             viewStartupParams = new ViewStartupParams(dynamoVM);
             DynamoSelection.Instance.Selection.CollectionChanged += OnSelectionCollectionChanged;
-
         }
 
         public void AddMenuItem(MenuBarType type, MenuItem menuItem, int index = -1)
         {
             AddItemToMenu(type, menuItem, index);
         }
 
+        /// <summary>
+        /// Adds the extension UI control element to a new tab in the extensions side bar.
+        /// </summary>
+        /// <param name="viewExtension">Instance of the view extension object that is being added to the extensions side bar.</param>
+        /// <param name="contentControl">Control UI element with a single piece of content of any type.</param>
+        /// <returns></returns>
+        internal void AddToExtensionsSideBar(IViewExtension viewExtension, ContentControl contentControl)
+        {
+            dynamoView.AddTabItem(viewExtension, contentControl);
```

Comment:
hmm, I see this call returns null or the tab that gets added, did you want to raise an exception or log to the console if no tab is added?

Reply:
Yes, added logging to the console.